packages/react-vapor/src/components/table-hoc/examples/TableHOCServerExample.tsx

```diff
@@ -0,0 +1,193 @@
+import * as $ from 'jquery';
+import * as moment from 'moment';
+import * as React from 'react';
+import {connect} from 'react-redux';
+import {RouteComponentProps, withRouter} from 'react-router';
+import * as _ from 'underscore';
+
+import {ExampleComponent} from '../../../../docs/src/components/ComponentsInterface';
+import {withServerSideProcessing} from '../../../hoc/withServerSideProcessing/withServerSideProcessing';
+import {IDispatch, IThunkAction} from '../../../utils/ReduxUtils';
+import {IReactVaporTestState} from '../../../utils/tests/TestUtils';
+import {LastUpdated} from '../../lastUpdated/LastUpdated';
+import {Section} from '../../section/Section';
+import {TableWithPaginationActions} from '../actions/TableWithPaginationActions';
+import {TableHeaderWithSort} from '../TableHeaderWithSort';
+import {TableHOC} from '../TableHOC';
+import {TableRowHeader} from '../TableRowHeader';
+import {TableRowNumberHeader} from '../TableRowNumberHeader';
+import {tableWithActions} from '../TableWithActions';
+import {tableWithBlankSlate} from '../TableWithBlankSlate';
+import {tableWithDatePicker} from '../TableWithDatePicker';
+import {tableWithFilter} from '../TableWithFilter';
+import {tableWithNewPagination} from '../TableWithNewPagination';
+import {tableWithPredicate} from '../TableWithPredicate';
+import {tableWithSort} from '../TableWithSort';
+import {tableWithUrlState} from '../TableWithUrlState';
+import {TableHOCExampleUtils} from '../utils/TableHOCExampleUtils';
+import {ITableHOCCompositeState, TableHOCUtils} from '../utils/TableHOCUtils';
+
+type TableHOCServerProps = RouteComponentProps & ReturnType<typeof mapDispatchToProps>;
+
+export interface IExampleRowData {
+    city: string;
+    email: string;
+
+    username: string;
+    dateOfBirth: Date;
+    id: string;
+}
+
+export interface IExampleServerTableState {
+    data: IExampleRowData[];
+    isLoading: boolean;
+}
+
+interface TableHOCServerExamplesState {
+    data: {users: [any]; count: number};
+    isLoading: boolean;
+}
+
+export const TableHOCServerExample: ExampleComponent = () => <TableHOCServer />;
+
+TableHOCServerExample.title = 'TableHOC server';
+
+// start-print
+export const TableHOCServerExampleId = 'complex-example';
+
+const renderHeader = (isLoading: boolean) => (
+    <thead>
+        <tr>
+            <TableRowNumberHeader isLoading={isLoading} />
+            <TableHeaderWithSort id="address.city" tableId={TableHOCServerExampleId} isLoading={isLoading}>

```

Comment:
Maybe its out of scope for now but I'm wondering if we could create a react context for the table in which we would put `tableId` and `isLoading`. Could be a future improvement

Reply:
added